### ISSUE TITLE
Mir does not support pen input devices

### ISSUE THREAD

**Issue by UniversalSuperBox** [MirServer/mir#1985]
Graphics tablets, x86 tablets, and many new and interesting devices are now shipping with a pressure- and tilt-sensitive active pen. These devices are recognized as Tablet Tools by libinput and passed to applications by Wayland compositors like KWin and Mutter. This allows users to use their graphics tablet as a relative input device, or their pen-enabled display as an absolute input device. These devices can draw in applications supporting pens, like [Krita](https://krita.org/en/) and [Xournal++](https://xournalpp.github.io/); and activate small controls more reliably than the capacitive sausage that is a finger.

While I am not entirely sure of the client-server dynamics of libinput and pen support under Wayland, I am fairly sure that Mir does not support active pens. I can use a pen device for pointing and clicking under GNOME Wayland on Ubuntu 20.04 on my Surface Go. When running egmde on the same device, nothing happens while the pen is in proximity or in contact with the screen.

**Comment by RAOF** [comment]
This *probably* needs the [tablet protocol](https://github.com/wayland-project/wayland-protocols/blob/master/unstable/tablet/tablet-unstable-v2.xml) implemented?

**Comment by UniversalSuperBox** [comment]
Yes, that seems correct. Qt appears to [implement this protocol](https://github.com/qt/qtwayland/commit/222455cd643c128fa9730c9c527a7fdcadd0acfe) already, and QtMir would need to implement it separately I'm sure.

**Comment by AlanGriffiths** [comment]
> QtMir would need to implement it separately

QtMir would need some logic to handle the events (in the same way it currently processes touch). The Wayland protocol would be implemented in Mir.

**Comment by Fuseteam** [comment]
fwiw the protocol is no longer unstable: https://wayland.app/protocols/tablet-v2
